### PR TITLE
Update tensorflow conda environment

### PR DESCRIPTION
**Pull request by christopherwoodall**
Update to fix audio poisoning scenarios. 

## Error
```
File "/workspace/armory/baseline_models/tf_graph/audio_resnet50.py", line 62, in train_step
model.optimizer.apply_gradients(zip(gradients, model.trainable_variables))
File "/opt/conda/lib/python3.9/site-packages/keras/optimizers/optimizer_experimental/optimizer.py", line 1140, in apply_gradients
return super().apply_gradients(grads_and_vars, name=name)
File "/opt/conda/lib/python3.9/site-packages/keras/optimizers/optimizer_experimental/optimizer.py", line 634, in apply_gradients
i
teration = self._internal_apply_gradients(grads_and_vars)
File "/opt/conda/lib/python3.9/site-packages/keras/optimizers/optimizer_experimental/optimizer.py", line 1166, in _internal_apply_gradients
return tf.__internal__.distribute.interim.maybe_merge_call(
File "/opt/conda/lib/python3.9/site-packages/keras/optimizers/optimizer_experimental/optimizer.py", line 1216, in _distributed_apply_gradients_fn
distribution.extended.update(
File "/opt/conda/lib/python3.9/site-packages/keras/optimizers/optimizer_experimental/optimizer.py", line 1211, in apply_grad_to_update_var
return self._update_step_xla(grad, var, id(self._var_key(var)))
Node: 'StatefulPartitionedCall_66'
libdevice not found at ./libdevice.10.bc
[[{{node StatefulPartitionedCall_66}}]] [Op:__inference_train_step_195557]
```

### CODE REVIEW

**Comment by christopherwoodall** [comment]
FYSA: Adding open [keras](https://github.com/keras-team/keras/issues/17422#issuecomment-1539388368) and [tensorflow](https://github.com/tensorflow/tensorflow/issues/56927) issues for tracking.

File: docker/Dockerfile-pytorch-deepspeech [hunk 1]
```diff
@@ -8,7 +8,9 @@ WORKDIR /armory-repo
 #       in the root of the repo.
 COPY ./ /armory-repo
 
-RUN pip install git+https://github.com/romesco/hydra-lightning/\#subdirectory=hydra-configs-pytorch-lightning
+RUN echo "Installing Hydra" && \
+    pip install hydra-core && \
+    pip install git+https://github.com/romesco/hydra-lightning/\#subdirectory=hydra-configs-pytorch-lightning
```
Comment (swsuggs) [inline docker/Dockerfile-pytorch-deepspeech:13]: What are we adding hydra for?

File: environment.yml [hunk 2]
```diff
@@ -5,7 +5,8 @@ channels:
   - conda-forge
   - defaults
 dependencies:
-  - pytorch=1.12
+  - tensorflow
+  - pytorch
```
Comment (swsuggs) [inline environment.yml:9]: I believe pytorch 2 will break parts of armory; is that restriction asserted anyplace else?

File: docker/Dockerfile-pytorch-deepspeech (hunk 1 above)
Comment (christopherwoodall) [reply docker/Dockerfile-pytorch-deepspeech]: During exploration, I found that the `deepspeech` container is missing the `hydra` requirement. An alternative location, that might make more sense looking back at it, would be in the `deepspeech` section of the `requirements`—code updated to reflect the previous statement.

File: environment.yml (hunk 2 above)
Comment (christopherwoodall) [reply environment.yml:9]: Updated pinned torch version in `pyproject.toml` to `torch >=1.12.0,<=1.13.1`.

File: docker/Dockerfile-pytorch-deepspeech (hunk 1 above)
Comment (swsuggs) [reply docker/Dockerfile-pytorch-deepspeech]: sounds good.

**Review by swsuggs — APPROVED** [review]
looks good, thanks for taking care of this Chris

**Pull request merged**